### ISSUE TITLE
How to return JSON array instead of text for specific column 

### ISSUE THREAD

**Issue by SamehMahmoud** [lightswitch05/table-to-json#29]
I want to return a JSON array from an extractor like this:

`

```
10 : function(cellIndex , $cell){
                    var prlArray = [];
                    $cell.find('select option').each(function(){
                        //alert(this.value);
                        prlArray.push(this.value);
                    });
                    return JSON.parse(prlArray.toString());
                }
```

`

This extractor is causing error in because of "return JSON.parse(prlArray.toString());"
thanks in advance
##

**Comment by lightswitch05** [comment]
does this not work?

```JavaScript
function(cellIndex , $cell){
    var prlArray = [];
    $cell.find('select option').each(function(){
        //alert(this.value);
        prlArray.push(this.value);
    });
    return prlArray);
}
```

**Comment by SamehMahmoud** [comment]
I does, but it is concatenated to the rest data as String not array like this :
{ parallelActions: "8,2,3"}
but what i need is to be like this:
{parallelActions : [8,2,3]}

**Comment by lightswitch05** [comment]
Just looked at the code. It will always convert whatever you give it into a string

**Comment by Mottie** [comment]
You could process `this.value` before pushing it into the `prlArray`.

Maybe you can try using this simple jQuery plugin called [`nVal()`](https://gist.github.com/Mottie/182adf51f05c932c53ff)... use `$(this).nVal()` and it will return an array of numbers.

**Comment by lightswitch05** [comment]
@Mottie I think the problem is that anything returned from the extractor function is automatically converted to a string by the `$.trim` function: https://github.com/lightswitch05/table-to-json/blob/master/src/jquery.tabletojson.js#L55

**Comment by koshuang** [comment]
@SamehMahmoud I @lightswitch05 I submitted a PR #37 which should be able to fix this issue.

**Comment by lightswitch05** [comment]
released in 0.12.0
